src/codem/preprocessing/preprocess.py

```diff
@@ -361,12 +365,15 @@ def _create_dsm(self, resample: bool = True) -> None:
         a scale factor to convert to meters.
         """
 
+        #might just have to project if in GCS.... but maybe there is a non-intrusive way to have the units and necessary scaling things converted to meters where they need to be.
+
         with rasterio.open(self.file) as data:
             resample_factor = (
                 self.native_resolution / self.resolution if resample else 1.0
             )
             tag = ["AOI", "Foundation"][int(self.fnd)]
-            if resample_factor != 1:
+            breakpoint()
+            if resample_factor != 1 or self.native_resolution != self.resolution:
```

Comment:
instead of using `!=` use `not math.isclose(self.native_resolution, self.resolution)` you may need to modify some of the default values for tolerance tho